### PR TITLE
Refactor/hooks

### PR DESCRIPTION
**Pull request by andrewvallejo**
# Hooks 
This PR was more quite the overhaul. I wanted to implement a global context to pass information easier and to have a single source of truth. I also didn't have functional logic inside of App so I separated one level deeper by creating page components. Every page's logic will now live inside of these components. All app has to do is render the component upon route change. I also disabled/removed error handling, loading, and a few features that would be better implemented with the global context. Everything visual is still the same as I want to revisit the wireframe. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactor

## Changes
- **Important Changes**
  - I am using Yarn instead of NPM due to its capabilities to clean up dependencies. There is no package.lock file.
  - App -> Pages -> Components   <-- This is the path in which things are rendered
- Remove useState from most components
- Added global context for App's state
- App state now holds more prevalent information 
- Refactor API calls to use Axios
- Add reducer to hold all actions for global context
- Remove PropTypes because all data types were changed during this refactor
- Removed loading & error handling (I want this to be much simpler)
- Add pages which is basically the hub for everything that lives on the page. It adds one level of complexity but I think it ultimately keeps everything more organized. Also, I know that global context can be accessed from anywhere. I am trying to only drill props from the root of the page component so that way all of the children components could still be dynamic just in case they needed to be used for something else.

## Concerns
**No concerns but more so the to-do list**
- Reintroduce useEffect to load questions. Currently, we are doing two fetch calls for the question and answers but I have a feeling we can rework the server and add that to the callback from the get request
- Not having error handling on the server made refactoring very difficult. I broke the server so many times (if this happens you have to restart Heroku)
- I want to add another row to the answer table of a boolean `hasVoted`  on the backend as well as have those checks in the front end. I don't wanna risk this feature getting abused. 
- I want to introduce my updated sass. 
- Redo wireframe
- Look into creating a user table/ user

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "axios": "^0.21.4",
```
Comment (rachaelcarroll) [inline package.json:12]: I've heard of axios but would like to look into it/go over it with you

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    (async () => !state.questions.length &&
+      await getQuestions().then(data => {
+        dispatch({ state, action: { type: 'SETQUESTIONS', value: data } })
+        dispatch({ state, action: { type: 'SETRANDOMQUESTION', value: randomize(data) } })
+      })
+    )()
+  }, [])
+
+  return (
+    <DevContext.Provider value={{ state, dispatch }}>
```
Comment (rachaelcarroll) [inline src/components/App.js:34]: guessing that this and the initial state is part of the global context

File: src/pages/HomePage.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, { useContext } from "react";
+import { QuestionForm } from "../components/QuestionForm";
+import { QuestionBoard } from "../components/QuestionBoard";
+import { DevContext } from "../utility/DevContext";
+
+export const HomePage = () => {
```
Comment (rachaelcarroll) [inline src/pages/HomePage.js:6]: nice refactor here. I did something similar in my mock take home challenge last week with a Dashboard component

File: src/utility/apiCalls.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import axios from "axios";
+
```
Comment (rachaelcarroll) [inline src/utility/apiCalls.js:2]: Did you use axios in any past projects? Im interested to hear what the benefits are

File: src/utility/reducer.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+export const reducer = (state, { action }) => {
+	const { type, value } = action
```
Comment (rachaelcarroll) [inline src/utility/reducer.js:2]: Also want to look into reducers. if you have any good resources/youtube vids please send my way :)

File: package.json (hunk 1 above)
Comment (andrewvallejo) [reply package.json:12]: I find it to be cleaner and easier than fetch calls. Also, postman shows you how you can write a HTTP request for all methods (XML, fetch, axios, etc)

File: src/components/App.js (hunk 2 above)
Comment (andrewvallejo) [reply src/components/App.js:34]: This is what gives you access to the state throughout the project!

File: src/utility/apiCalls.js (hunk 4 above)
Comment (andrewvallejo) [reply src/utility/apiCalls.js:2]: This was the second time I used it. The first was in the Mondo Robot project. It gets rid of a lot of the nuances that fetch has such as no stringifying the body, no needing request.ok, etc.

[fetch vs axios](https://www.geeksforgeeks.org/difference-between-fetch-and-axios-js-for-making-http-requests/)

**Pull request merged**